Code/Tools/AssetProcessor/native/tests/assetmanager/AssetProcessorManagerTest.cpp

```diff
@@ -4241,8 +4257,9 @@ TEST_F(LockedFileTest, DeleteFile_LockedProduct_DeletesWhenReleased)
 
         EXPECT_FALSE(QFile::exists(productPath));
         EXPECT_EQ(m_data->m_deletedSources.size(), 1);
-
-        workerThread.join();
+        
+        EXPECT_GT(m_deleteCounter, 1); // Make sure the AP tried more than once to delete the file
+        m_errorAbsorber->ExpectAsserts(0);
     }
     else
     {
```

Comment:
just to be safe I would suggest putting a `m_callback = {}` at the end of this method
since the callback is using a variable that is in the stack, it would be really dangerous if for some reason it gets called after this method (e.g. during teardown)

Reply:
In the code that calls the callback, I've set it up to clear the callback to be sure its never called again

Reply:
I would still add it here just to tie the lifetime of that callback to this method
If for some reason that callback is not called during the method (e.g. if the switch/if dont catch the message) there is a potential for another call to call it after.
Since these tests are playing to get the right events triggered at exactly the right time, I would opt on the safe side.